File: src/main/java/com/refinedmods/refinedstorage/container/factory/CraftingMonitorContainerFactory.java
package com.refinedmods.refinedstorage.container.factory;

import com.refinedmods.refinedstorage.RSContainerMenus;
import com.refinedmods.refinedstorage.container.CraftingMonitorContainerMenu;
import com.refinedmods.refinedstorage.blockentity.craftingmonitor.CraftingMonitorBlockEntity;
import net.minecraft.core.BlockPos;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.world.entity.player.Inventory;
import net.minecraftforge.network.IContainerFactory;

public class CraftingMonitorContainerFactory implements IContainerFactory<CraftingMonitorContainerMenu> {
    @Override
    public CraftingMonitorContainerMenu create(int windowId, Inventory inv, FriendlyByteBuf data) {
        BlockPos pos = data.readBlockPos();

        CraftingMonitorBlockEntity blockEntity = (CraftingMonitorBlockEntity) inv.player.level.getBlockEntity(pos);

        return new CraftingMonitorContainerMenu(RSContainerMenus.CRAFTING_MONITOR.get(), blockEntity.getNode(), blockEntity, inv.player, windowId);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/factory/CraftingMonitorContainerFactory.kt
package com.refinedmods.refinedstorage.container.factory

import com.refinedmods.refinedstorage.RSContainers
import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import com.refinedmods.refinedstorage.container.CraftingMonitorContainer
import com.refinedmods.refinedstorage.tile.craftingmonitor.CraftingMonitorTile
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.network.PacketByteBuf
import net.minecraft.util.math.BlockPos
import net.minecraftforge.fml.network.IContainerFactory

class CraftingMonitorContainerFactory : IContainerFactory<CraftingMonitorContainer?> {
    fun create(windowId: Int, inv: PlayerInventory, data: PacketByteBuf): CraftingMonitorContainer {
        val pos: BlockPos = data.readBlockPos()
        val tile = inv.player.world.getBlockEntity(pos) as CraftingMonitorTile?
        return CraftingMonitorContainer(RSContainers.CRAFTING_MONITOR, tile.getNode(), tile, inv.player, windowId)
    }
}